File: src/tools/information.ts
import { type ToolRegistrar, jsonResult } from "./context.js";

export const registerInformationTools: ToolRegistrar = (server, ctx) => {
  server.tool(
    "get_store_info",
    "Get store account and server information: store id, name, domain, currency, " +
      "game type, online mode, and linked server details. " +
      "Use this to verify the connection and discover store configuration.",
    {},
    async () => {
      const data = await ctx.client.getInformation();
      return jsonResult(data);
    },
  );
};

File: src/tebex_mcp/tools/information.py
"""Store information tool."""

from __future__ import annotations

from typing import Any

from fastmcp import FastMCP

from tebex_mcp.context import ToolContext
from tebex_mcp.tools._common import map_tebex_errors


def register(mcp: FastMCP, ctx: ToolContext) -> None:
    @mcp.tool(
        name="get_store_info",
        description="Get store and server info: id, name, domain, currency, game type, online mode.",
    )
    @map_tebex_errors
    async def get_store_info() -> Any:
        return await ctx.client.get_information()
